save-backend/src/main/kotlin/org/cqfn/save/backend/service/ProjectService.kt

```diff
@@ -1,33 +1,50 @@
 package org.cqfn.save.backend.service
 
 import org.cqfn.save.backend.repository.ProjectRepository
+import org.cqfn.save.backend.repository.UserRepository
 import org.cqfn.save.domain.ProjectSaveStatus
 import org.cqfn.save.entities.Project
 import org.cqfn.save.entities.ProjectStatus
 import org.springframework.data.domain.Example
 import org.springframework.data.domain.ExampleMatcher
+import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Service
+import org.springframework.web.server.ResponseStatusException
 
 /**
  * Service for project
  *
  * @property projectRepository
  */
 @Service
-class ProjectService(private val projectRepository: ProjectRepository) {
+class ProjectService(private val projectRepository: ProjectRepository,
+                     private val userRepository: UserRepository,
+) {
     /**
      * Store [project] in the database
      *
      * @param project a [Project] to store
+     * @param username name of the user that should be associated as a creator of this project. If null, [project] should contain valid user id.
      * @return project's id, should never return null
      */
-    fun saveProject(project: Project): Pair<Long, ProjectSaveStatus> {
+    @Suppress("UnsafeCallOnNullableType")
+    fun saveProject(project: Project, username: String?): Pair<Long, ProjectSaveStatus> {
         val exampleMatcher = ExampleMatcher.matchingAll()
             .withMatcher("name", ExampleMatcher.GenericPropertyMatchers.exact())
             .withMatcher("owner", ExampleMatcher.GenericPropertyMatchers.exact())
         val (projectId, projectSaveStatus) = projectRepository.findOne(Example.of(project, exampleMatcher)).map {
             Pair(it.id, ProjectSaveStatus.EXIST)
         }.orElseGet {
+            username?.let {
+                userRepository.findByName(username)
+                    .map {
+                        project.userId = it.id!!
```

Comment:
This is incorrect; userId from frontend is -1 in this case, because frontend knows only user name, and `Project` instance is not completed.